networkx/utils/backends.py

```diff
@@ -1862,21 +1873,29 @@ def _convert_graph(
                 return graph
             backend = _load_backend(graph.__networkx_backend__)
             rv = backend.convert_to_nx(graph)
+            func = backend.convert_to_nx
+            kwargs = {}
         else:
             backend = _load_backend(backend_name)
-            rv = backend.convert_from_nx(
-                graph,
-                edge_attrs=edge_attrs,
-                node_attrs=node_attrs,
-                preserve_edge_attrs=preserve_edge_attrs,
-                preserve_node_attrs=preserve_node_attrs,
+            func = backend.convert_from_nx
+            kwargs = {
+                "edge_attrs": edge_attrs,
+                "node_attrs": node_attrs,
+                "preserve_edge_attrs": preserve_edge_attrs,
+                "preserve_node_attrs": preserve_node_attrs,
                 # Always preserve graph attrs when we are caching b/c this should be
                 # cheap and may help prevent extra (unnecessary) conversions. Because
                 # we do this, we don't need `preserve_graph_attrs` in the cache key.
-                preserve_graph_attrs=preserve_graph_attrs or use_cache,
-                name=self.name,
-                graph_name=graph_name,
-            )
+                "preserve_graph_attrs": preserve_graph_attrs or use_cache,
+                "name": self.name,
+                "graph_name": graph_name,
+            }
+        try:
+            rv = func(graph, **kwargs)
+        except Exception:
+            if use_cache and nx_cache is not None:
```

Comment:
Is `use_cache` ever used without `and nx_cache is not None`?
Maybe at the start of the function where `nx_cache` is created we could set `use_cache = use_cache and nx_cache is not None`

Reply:
Good suggestion. I cleaned it up :+1: